sympy/stats/tests/test_finite_rv.py

```diff
@@ -245,6 +245,37 @@ def test_binomial_symbolic():
     Y = Binomial('Y', n, p, succ=H, fail=T)
     assert simplify(E(Y) - (n*(H*p + T*(1 - p)))) == 0
 
+def test_beta_binomial_verify_parameters():
+    raises(ValueError, lambda: BetaBinomial('b', .2, 1, 2))
+    raises(ValueError, lambda: BetaBinomial('b', 2, -1, 2))
+    raises(ValueError, lambda: BetaBinomial('b', 2, 1, -2))
+
+def test_beta_binomial_numeric():
+    nvals = range(1,5)
+    alphavals = [S(1)/4, S.Half, S(3)/4, 1, 10]
+    betavals = [S(1)/4, S.Half, S(3)/4, 1, 10]
+
+    for n in nvals:
+        for a in alphavals:
+            for b in betavals:
+                X = BetaBinomial('X', n, a, b)
+                assert E(X) == moment(X, 1)
+                assert variance(X) == cmoment(X, 2)
+
+def test_beta_binomial_symbolic():
+    n = 2 # Because we're using for loops, can't do symbolic n
+    a, b = symbols('a b', positive=True)
+    X = BetaBinomial('X', n, a, b)
+    t = Symbol('t')
+
+    assert simplify(E(X)) == simplify(moment(X, 1))
+    assert simplify(variance(X)) == simplify(cmoment(X, 2))
+    assert skewness(X) == smoment(X, 3)
+    assert characteristic_function(X)(t) == exp(2*I*t)*beta(a + 2, b)/beta(a, b) +\
+         2*exp(I*t)*beta(a + 1, b + 1)/beta(a, b) + beta(a, b + 2)/beta(a, b)
+    assert moment_generating_function(X)(t) == exp(2*t)*beta(a + 2, b)/beta(a, b) +\
+         2*exp(t)*beta(a + 1, b + 1)/beta(a, b) + beta(a, b + 2)/beta(a, b)
+
```

Comment:
Would it be cool to write `symbolic` and `numeric` tests under the same `test` function?

Reply:
Done.